File: Lab/SEM4/MPP/lab4/moto_c/repository/RacerRepository.cs
using moto_c.entity;

namespace moto_c.repository;

public interface RacerRepository
{
    Racer? find(long id);
    
    List<Racer> getAll();

    List<Racer> findBy(Team team);

    List<Racer> findBy(RaceEvent engine);
    
    void add(Racer racer);
    
    void modify(Racer racer);
}

File: Lab/SEM4/MPP/lab4/moto_java/src/main/java/org/zdroba/repository/RacerRepository.java
package org.zdroba.repository;

import org.zdroba.entity.RaceEvent;
import org.zdroba.entity.Racer;
import org.zdroba.entity.Team;

import java.util.List;

public interface RacerRepository {
    Racer find(Long id);

    List<Racer> getAll();

    List<Racer> getBy(Team team);

    List<Racer> getBy(RaceEvent engine);

    void add(Racer racer);

    void modify(Racer racer);
}
